go/libraries/doltcore/sqle/enginetest/dolt_queries.go

```diff
@@ -565,6 +565,408 @@ var DoltMerge = []enginetest.ScriptTest{
+				Expected: []sql.Row{
+					{100, 101, nil, nil, "added"},
+				},
+			},
+			{
+				// TODO: What's the best way to test show warnings? With a custom/non-scripted test?
+				//       script_queries provides ExpectedWarning, but still expects you to provide expected sql.Rows
+				// TODO: SQL console shows this as four warnings, but the test framework is returning 8
+				Query: "show warnings;",
+				Expected: []sql.Row{
+					{"Warning", 1105, "unable to coerce value from field 'c' into latest column schema"},
+					{"Warning", 1105, "unable to coerce value from field 'c' into latest column schema"},
+					{"Warning", 1105, "unable to coerce value from field 'c' into latest column schema"},
+					{"Warning", 1105, "unable to coerce value from field 'c' into latest column schema"},
+					{"Warning", 1105, "unable to coerce value from field 'c' into latest column schema"},
+					{"Warning", 1105, "unable to coerce value from field 'c' into latest column schema"},
+					{"Warning", 1105, "unable to coerce value from field 'c' into latest column schema"},
+					{"Warning", 1105, "unable to coerce value from field 'c' into latest column schema"},
+				},
+			},
+		},
+	},
+	{
+		Name: "multiple column renames",
+		SetUpScript: []string{
+			"create table t (pk int primary key, c1 int);",
+			"insert into t values (1, 2);",
+			"set @Commit1 = (select DOLT_COMMIT('-am', 'creating table t'));",
+
+			"alter table t rename column c1 to c2;",
+			"insert into t values (3, 4);",
+			"set @Commit2 = (select DOLT_COMMIT('-am', 'renaming c1 to c2'));",
+
+			"alter table t drop column c2;",
+			"set @Commit3 = (select DOLT_COMMIT('-am', 'dropping column c2'));",
+
+			"alter table t add column c2 int;",
+			"insert into t values (100, '101');",
+			"set @Commit4 = (select DOLT_COMMIT('-am', 'recreating column c2'));",
+		},
+		Assertions: []enginetest.ScriptTestAssertion{
+			{
+				Query:    "SELECT COUNT(*) FROM DOLT_DIFF_t;",
+				Expected: []sql.Row{{5}},
+			},
+			{
+				Query: "SELECT to_pk, to_c2, from_pk, from_c2, diff_type FROM DOLT_DIFF_t WHERE TO_COMMIT=@Commit1 ORDER BY to_pk;",
+				Expected: []sql.Row{
+					{1, nil, nil, nil, "added"},
+				},
+			},
+			{
+				Query: "SELECT to_pk, to_c2, from_pk, from_c2, diff_type FROM DOLT_DIFF_t WHERE TO_COMMIT=@Commit2 ORDER BY to_pk;",
+				Expected: []sql.Row{
+					{3, 4, nil, nil, "added"},
+				},
+			},
+			{
+				Query: "SELECT to_pk, to_c2, from_pk, from_c2, diff_type FROM DOLT_DIFF_t WHERE TO_COMMIT=@Commit3 ORDER BY to_pk;",
+				Expected: []sql.Row{
+					{1, nil, 1, 2, "modified"},
+					{3, nil, 3, 4, "modified"},
+				},
+			},
+			{
+				Query: "SELECT to_pk, to_c2, from_pk, from_c2, diff_type FROM DOLT_DIFF_t WHERE TO_COMMIT=@Commit4 ORDER BY to_pk;",
+				Expected: []sql.Row{
+					{100, 101, nil, nil, "added"},
+				},
+			},
+		},
+	},
+	{
+		Name: "primary key change",
```

Comment:
These are great tests, but you're missing a large and important bit of surface area: asserting that `dolt_diff AS OF @commit1` has the schema the table at that commit. And should include data changes after that schema change, but in terms of the old schema.

Reply:
Interesting... should `select * from dolt_diff_$table as of @Commit1` work? I tried this while testing and skipped it because it didn't seem like it was supported. I just tested with my non-dev dolt build from homebrew (version 0.35.10) and couldn't get as of to work with `dolt_diff_$table` either. 

I had assumed that using `as of @Commit1` would work as if @Commit1 were checked out – i.e. none of the diff data more recent than @Commit1 would be included. It sounds like you think it should just use the old schema version, but try to fit all the current data up to HEAD into it?

Reply:
Right, the AS OF clause only specifies the schema for the diff, the diff table's data always has the entire diff history of the repository. Otherwise you can only work backwards from an older schema, which is very limiting.

Reply:
Gotcha. Currently no system tables support `as of`. I'm happy to pick up that work – I see how the `GetTableInsensitiveAsOf` code is going around the code where we instantiate system tables. I'd prefer to tackle that in a separate PR if you're cool with that though.

Reply:
Chatted with Zach yesterday and confirmed this plan – I'll tackle `as of` support for system tables in a separate PR and come back to covering those cases in these tests.